pynecone/state.py

```diff
@@ -61,6 +61,39 @@ def __init__(self, *args, **kwargs):
         for substate in self.get_substates():
             self.substates[substate.get_name()] = substate().set(parent_state=self)
 
+        self._init_mutable_fields()
+
+    def _init_mutable_fields(self):
+        """Initialize mutable fields.
+
+        So that mutation to them can be detected by the app:
+        * list
+        """
+        for field in self.base_vars.values():
+            if field.type_ is list:
```

Comment:
We can use `utils._issubclass(field.type_, List)` here which will work for things like `typing.List[int]` also.

Also let's use `List` everywhere because we need to support Python 3.7+